src/open_samus_returns_rando/files/templates/randomizerpowerup.lua

```diff
@@ -96,6 +97,34 @@ function RandomizerPowerup.MarkLocationCollected(actorOrName)
     Blackboard.SetProp(playerSection, propName, "b", true)
 end
 
+local AeionAbilityClouds = {
+    {"s000_surface", "LE_SpecialAbility_ScanningPulse"},
+    {"s028_area2c", "LE_SpecialAbility_EnergyShield"},
+    {"s030_area3", "LE_SpecialAbility_EnergyWave"},
+    {"s060_area6", "LE_SpecialAbility_PhaseDisplacement"},
```

Comment:
To me it looks like the names are unique, so there is no reason to have the area name + the check for it IMO.

Reply:
I'm doing it this way because all of the aeion ability clouds are named the same, so getting one of these would activate the rest.

Reply:
Hmmm but it doesn't activate it in a specific scenario, it's always just
```lua
SpecialEnergyCloud.ActivateSpecialEnergy(cloud)
```
where `cloud = "TG_SpecialEnergyCloud"`.
This only activates it in the scenario you are currently at.

Reply:
That's what the loop is for. If the scenarios don't match then it doesn't activate in the scenario

Reply:
Yes, but my point is that you will never find e.g. `"LE_SpecialAbility_ScanningPulse"` in a different scenario than `"s000_surface"`, so checking the scenario is not relevant.

Reply:
I guess I'll test without the check but i don't think it's gonna work

Reply:
I try it with other words:
The loop is currently:
```lua
    for _, mapping in ipairs(AeionAbilityClouds) do
        if mapping[1] == Scenario.CurrentScenarioID and mapping[2] == name then
            SpecialEnergyCloud.ActivateSpecialEnergy(cloud)
        end
    end
```
`name` is the name of the actor you just picked up and `mapping[2]` is one of the values `"LE_SpecialAbility_ScanningPulse", "LE_SpecialAbility_EnergyShield", "LE_SpecialAbility_EnergyWave", "LE_SpecialAbility_PhaseDisplacement"`

So, if you just picked up `LE_SpecialAbility_PhaseDisplacement` there is no way that the check  ` mapping[2] == name` is `true` without being in `s060_area6` because that actor name doesn't exist in any other scenario.
For all other pickups it's `false` because the name is gonna be something like `LE_Item_001` etc.

Reply:
Couldn't I just leave the checks but modify the name check to be get entity?

Reply:
For what purpose?

Reply:
This is me just talking what might happen but I just need to test and see.